File: 3Arrays/Array_operation code/Insertion/insertion.java
public class insertion{
    public static void display(int arr[],int n){
        for(int i=0;i<n;i++){
            System.out.println(arr[i]+" ");
        }
        System.out.println();
    }
    public static int Insertion(int arr[],int capacity,int index,int size,int element){
        if(size>=capacity){
            return -1;
        }
        for(int i=size-1;i>=index;i--){
            arr[i+1]=arr[i];
        }
         arr[index]=element;
         return 1;
    }
    public static void main(String[] args) {
        int []arr=new int[100];
        arr[0]=56;
        arr[1]=67;
        arr[2]=58;
        arr[3]=89;
        arr[4]=56;
        int capacity=arr.length;
        int size=5;
        int index=2;
        int element=45;
        display(arr,size);
        int flag=Insertion(arr,capacity,index,size,element);
        size+=1;
        if(flag==1){
            display(arr,size);
            System.out.println("the"+element+"is inserted at "+index);
        }
        else{
            System.out.println("The array is full");
        }
    }
}

File: 3Arrays/Array_operation code/Insertion/insertion.c
#include <stdio.h>
//insertion Function
// int  indInsertion(int arr[],int size,int element,int capacity,int index){
//     if(size>=capacity){
//         printf("The array is full");
//         return -1;
//     }
//     for(int i=size-1;i>=index;i--)
//     {
//         arr[i+1]=arr[i];
//     }
//     arr[index]=element;
//     return 1;
// }
int indInsertion(int arr[],int size,int element,int capacity,int index){
    if(size>=capacity){
        // printf("The array is Full");
        return -1;
    }
    for(int i=size-1;i>=index;i--){
        arr[i+1]=arr[i];//5=4,4=3,
    }
    arr[index]=element;
    return 1;
}
void display(int arr[],int n){
    for(int i=0;i<n;i++){
        printf("%d",arr[i]);
    }
    printf("\n");
}
int main(){
    int arr[100]={7,8,12,27,88};
    int size=5;
    int element=45;
    int index=3;
    display(arr,size);
    int flag=indInsertion(arr,size,element,100,index);
    size +=1;
    if(flag==1){
        display(arr,size);
        printf("the element is inserted at index %d",index);
    }
    else{
        printf("The array is full");
    }
  return 0;
}